Utilities/bootstrap

```diff
@@ -892,9 +892,19 @@ exec ${{__dir}}/swift-$1 ${{SPM_FLAGS[@]}} "${{@:2}}" """.format(options=swiftpm
     st = os.stat(script_path)
     os.chmod(script_path, st.st_mode | stat.S_IEXEC)
 
+def checkTool(cmd):
+    # Confirm cmake is available.
+    try:
+        subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+    except Exception as e:
+        error(cmd[0]+" is not available: %s" % e) 
+
 def build_llbuild(args):
     note("Building llbuild")
-
+    
+    checkTool(["ninja", "--version"])    
+    checkTool(["cmake", "--version"])
```

Comment:
we should also use this new method for rsync check in the main method

Reply:
done